crates/gpui/src/platform/windows/platform.rs

```diff
@@ -204,25 +210,28 @@ impl Platform for WindowsPlatform {
 
     fn run(&self, on_finish_launching: Box<dyn 'static + FnOnce()>) {
         on_finish_launching();
-        loop {
-            match self.wait_message() {
-                WindowsMessageWaitResult::ForegroundExecution => {
-                    for runnable in self.inner.main_receiver.drain() {
-                        runnable.run();
-                    }
-                }
-                WindowsMessageWaitResult::WindowsMessage(msg) => {
+        'a: loop {
+            let mut msg = MSG::default();
+            let wait_result =
+                unsafe { DCompositionWaitForCompositorClock(Some(&[self.inner.event]), INFINITE) };
+
+            unsafe { invalidate_thread_windows(GetCurrentThreadId()) };
+
+            if wait_result == 1 {
```

Comment:
What is this `1` mean? I read `DCompositionWaitForCompositorClock` document, but only `Type: DWORD` is written in return value section.

Reply:
yea the docs are really sparce for that one. here's a more detailed doc https://learn.microsoft.com/en-us/windows/win32/directcomp/compositor-clock/compositor-clock

Here's relevant part: 
> If we get WAIT_RESULT_0+count it means the compositor clock ticked,
and we should render another frame. Our count is one, as we're
passing only one extra handle. Otherwise, either we got a failure or
another thread signaled our "quit" event, and in either case we want
to exit the loop

`WAIT_RESULT_0+count` would be `1` in this case

Reply:
Then we should invalidate window iff `wait_result == 1`?

Oh I found this:

> This is critical for power management, thus restricting the architecture of the API to be a wait function call, rather than accepting or returning an event (the graphics system can't determine whether or not it's being waited on). At this low level, applications are expected to use this API to control rendering threads, separate from general-purpose UI threads, similar to how IDXGIOutput::WaitForVBlank is traditionally used.

Maybe this function blocks window messages, and prefer to use this on background thread.

Reply:
Uh, I meant creating a rendering thread that is synchronized with the refresh rate, rather than using the existing background thread that the Dispatcher is running on.

For example, [the Unreal Engine separates the rendering thread from the UI thread](https://docs.unrealengine.com/4.26/en-US/ProgrammingAndScripting/Rendering/ThreadedRendering/), and I believe this is a common practice.

Maybe @kvark knows more about graphics, let me know if you have any pointers.

Reply:
im all for trying a different approach, but for now I don't think this should be a blocker for this PR.

From asking on discord @mikayla-maki was mentioning that the request frame callback should run at the refresh rate (https://discord.com/channels/869392257814519848/869392258292645892/1214655841073111171) which is what the mac platform does with [CADisplayLink](https://developer.apple.com/documentation/quartzcore/cadisplaylink). This PR achieves that which is what GPUI wants AFAIK.

Reply:
Could we include the documentation for why this is 1 in a comment in the PR? As well as another comment in the location that makes this '1'?